### PR TITLE
Included docs, changelogs, authors and fixed some typos and errors

### LINKED ISSUE #2252: Update contribution guidelines to include doc, changelogs and authors
This would help keeping these up to date!

### PR DESCRIPTION
**Pull request by shricodev**
I have added a section down the file to include docs, changelogs and authors to keep things updated and also fixed some of the grammatical errors and typos in the CONTRIBUTING.rst file 

Signed-off-by: Shrijal Acharya coveringguy0123@gmail.com

Fixes #2252

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in a uniquely-named feature branch and have no merge conflicts 📁

### CODE REVIEW

**Comment by pombredanne** [comment]
Thanks... can you review our commit message guidelines?
Also for this PR title instead of "Included all the details to the file and fixed some typos and grammatical errors." ... can you do better?

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -14,16 +14,15 @@ check `easy issues <https://github.com/nexB/scancode-toolkit/labels/easy>`_
...
 
 For commits, it is best to simply add a line like this to your commit message,
 with your name and email::
 
     Signed-off-by: Jane Doe <developer@example.com>
 
-Please try to write a good commit message, see `good commit message wiki
-<https://aboutcode.readthedocs.io/en/latest/contributing/writing_good_commit_messages.html>` for
-details. In particular use the imperative for your commit subject: think that
+Please try to write a good commit message, see `good commit message wiki <https://aboutcode.readthedocs.io/en/latest/contributing/writing_good_commit_messages.html>`_ for
```
Comment (pombredanne) [inline CONTRIBUTING.rst]: Not sure why you created long lines?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -14,16 +14,15 @@ check `easy issues <https://github.com/nexB/scancode-toolkit/labels/easy>`_
...
+details. In particular use, the imperative for your commit subject: think that
```
Comment (pombredanne) [inline CONTRIBUTING.rst]: The comma looks out of place

File: CONTRIBUTING.rst (hunk 1 above)
Comment (pombredanne) [reply CONTRIBUTING.rst]: Also please read the commit message doc at this URL

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -228,3 +227,20 @@ To run tests in parallel on eight processors::
 To run tests verbosely, displaying all print statements to terminal::
 
     py.test  -vvs
+
+Things to Keep in Mind When Contributing
+----------------------------------------
+
+#. An update to the ``CHANGELOG`` is required if any important changes are made that needs to be communicated such as:
```
Comment (AyanSinhaMahapatra) [inline CONTRIBUTING.rst]: This shouldn't be in a different section, instead include this in https://github.com/nexB/scancode-toolkit/blob/70564a4a513d46eaa1fc0deff1cf1b61200df940/CONTRIBUTING.rst#development between points 5 and 6.

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -146,7 +145,7 @@ To set up ScanCode for local development:
 
         or
 
-        . venv/bin/activate
+        .venv/bin/activate
```
Comment (pombredanne) [inline CONTRIBUTING.rst]: This is wrong. Try it out and you will see

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks. Here are some comments!

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -228,3 +227,20 @@ To run tests in parallel on eight processors::
...
+   
+   * Addition or deletion of CLI options.
+   
+   * Addition of any new feature to the program.
+   
+   * Any other miscellaneous changes to the program.
+   
+#. If there is a code change, a significant document, or any other changes, you must update the ``AUTHORS`` to include your own name.
+
+#. A link to the related documentation, which contains a more detailed guide for contributing code or documentation.
```
Comment (AyanSinhaMahapatra) [inline CONTRIBUTING.rst]: Same, include this between points 3 and 4, and this time actually include the link to the documentation for contributing code: https://scancode-toolkit.readthedocs.io/en/stable/contribute/contrib_dev.html

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
Thanks, see comments. These should not be a new section, but new points included within.

**Comment by shricodev** [comment]
@pombredanne  @AyanSinhaMahapatra Let me know what you think. In case you need any changes will do it asap!

File: CONTRIBUTING.rst (hunk 2 above)
Comment (shricodev) [reply CONTRIBUTING.rst]: Fixed.

File: CONTRIBUTING.rst (hunk 4 above)
Comment (shricodev) [reply CONTRIBUTING.rst]: Fixed.

File: CONTRIBUTING.rst (hunk 3 above)
Comment (shricodev) [reply CONTRIBUTING.rst]: Added. Can you look into the changes?😀

File: CONTRIBUTING.rst (hunk 5 above)
Comment (shricodev) [reply CONTRIBUTING.rst]: Added. Can you look into the changes?

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -160,6 +160,16 @@ To set up ScanCode for local development:
 5. Now you can make your code changes in your local clone.
    Please create new unit tests for your code. We love tests!
 
+   An update to the ``CHANGELOG`` is required if any important changes are made that needs to be communicated such as:
```
Comment (AyanSinhaMahapatra) [inline CONTRIBUTING.rst:163]: This should have a new point number. Also update the others accordingly.

File: CONTRIBUTING.rst [hunk 7]
```diff
@@ -124,11 +122,13 @@ To set up ScanCode for local development:
 3. Create a branch for local development::
 
     git checkout -b name-of-your-bugfix-or-feature
+
```
Comment (AyanSinhaMahapatra) [inline CONTRIBUTING.rst:125]: This should have a point number, adjust others accordingly. 

Also something like `check out the contributing to developement documentation, as it contains ...` may be better for this?

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
See comments.

Also can you squash and force push the commits? DCO is failing as you don't have a signoff in the previous commits and all of this can in be a single commit IMHO

See https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History for reference

**Comment by shricodev** [comment]
@AyanSinhaMahapatra I seem to have messed up my local copy of the repository. I am closing this and opening a new PR for the same. I am really sorry for the inconvenience caused.

**Pull request closed without merging**